AllJoyn/Platform/BridgeHostStartupTask/Program.cs

```diff
@@ -0,0 +1,140 @@
+        }
+
+        /// <summary>
+        /// Creates the app service connection
+        /// </summary>
+        static async void ThreadProc()
+        {
+            while (true)
+            {
+                if (isConnectionClosed ||
+                    !(await isHealthy()))
+                {
+                    connection = new AppServiceConnection();
+                    connection.AppServiceName = "CommunicationService";
+                    connection.PackageFamilyName = "OpenT2TAllJoynBridgeHost_a6049pvczh34w";
+                    isConnectionClosed = false;
+                    AppServiceConnectionStatus status = await connection.OpenAsync();
+                    switch (status)
+                    {
+                        case AppServiceConnectionStatus.Success:
+                            Console.ForegroundColor = ConsoleColor.Green;
+                            connection.RequestReceived += Connection_RequestReceived;
+                            connection.ServiceClosed += Connection_ServiceClosed;
+                            Console.WriteLine("Connection established - waiting for requests");
+                            Console.WriteLine();
+                            break;
+                        case AppServiceConnectionStatus.AppNotInstalled:
+                            Console.ForegroundColor = ConsoleColor.Red;
+                            Console.WriteLine("The app AppServicesProvider is not installed.");
+                            break;
+                        case AppServiceConnectionStatus.AppUnavailable:
+                            Console.ForegroundColor = ConsoleColor.Red;
+                            Console.WriteLine("The app AppServicesProvider is not available.");
+                            break;
+                        case AppServiceConnectionStatus.AppServiceUnavailable:
+                            Console.ForegroundColor = ConsoleColor.Red;
+                            Console.WriteLine(string.Format("The app AppServicesProvider is installed but it does not provide the app service {0}.", connection.AppServiceName));
+                            break;
+                        case AppServiceConnectionStatus.Unknown:
+                            Console.ForegroundColor = ConsoleColor.Red;
+                            Console.WriteLine(string.Format("An unkown error occurred while we were trying to open an AppServiceConnection."));
+                            break;
+                    }
+                }
+
+                Thread.Sleep(5000); // Do Health checks every 5 seconds
+            } // while()
+        }
+
+        private static void Connection_ServiceClosed(AppServiceConnection sender, AppServiceClosedEventArgs args)
+        {
+            Console.ForegroundColor = ConsoleColor.Red;
+            isConnectionClosed = true;
+            connection = null;
+            Console.WriteLine(string.Format("AppServiceConnection is closed!: Status : " + args.Status));
```

Comment:
Actually nullifying is not required. I shall remove that and keep it simple.

---
In reply to: [75188785](https://github.com/tjaffri/msiot-samples/pull/9#discussion_r75188785) [](ancestors = 75188785)